src/commands.ts

```diff
@@ -52,17 +58,70 @@ function bzlMakeQueryQuickPickItem(queryItem: BazelQueryItem): BazelQueryQuickPi
 }
 
 async function bzlQuickPickQuery(query: string = '...', options?: QuickPickOptions) {
+
     return Window.showQuickPick(bzlQuery(query).then(deps => {
         return deps.map(bzlMakeQueryQuickPickItem);
-    }, err => {
-        Window.showQuickPick([], {
+    }).catch(async err => {
+        await Window.showQuickPick([], {
             placeHolder: "<ERROR>"
-        })
-        Window.showErrorMessage(err.toString())
+        });
         return [];
     }), options);
 }
 
+function bzlClearQueryDiagnostics() {
+    bazelDiagnosticsCollection.clear();
+}
+
+function bzlQueryErrorDiagnostics(error: Error) {
```

Comment:
A bit more documentation would be nice. Can we describe the error output in some kind of parser rules?